iib/workers/tasks/utils.py

```diff
@@ -68,6 +73,80 @@ def inner(*args, **kwargs):
     return wrapper
 
 
+def reset_docker_config():
+    """Create a symlink from ``iib_docker_config_template`` to ``~/.docker/config.json``."""
+    conf = get_worker_config()
+    docker_config_path = os.path.join(os.path.expanduser('~'), '.docker', 'config.json')
+
+    try:
+        log.debug('Removing the Docker config at %s', docker_config_path)
+        os.remove(docker_config_path)
+    except FileNotFoundError:
+        pass
+
+    if os.path.exists(conf.iib_docker_config_template):
+        log.debug(
+            'Creating a symlink from %s to %s', conf.iib_docker_config_template, docker_config_path
+        )
+        os.symlink(conf.iib_docker_config_template, docker_config_path)
+
+
+@contextmanager
+def set_registry_token(token, container_image):
+    """
+    Configure authentication to the registry that ``container_image`` is from.
+
+    This context manager will reset the authentication to the way it was after it exits. If
+    ``token`` is falsy, this context manager will do nothing.
+
+    :param str token: the token in the format of ``username:password``
+    :param str container_image: the pull specification of the container image to parse to determine
+        the registry this token is for.
+    :return: None
+    :rtype: None
+    """
+    if not token:
+        log.debug(
+            'Not changing the Docker configuration since no overwrite_from_index_token was provided'
+        )
+        yield
+
+        return
+
+    docker_config_path = os.path.join(os.path.expanduser('~'), '.docker', 'config.json')
+    try:
+        log.debug('Removing the Docker config symlink at %s', docker_config_path)
+        try:
+            os.remove(docker_config_path)
+        except FileNotFoundError:
+            log.debug('The Docker config symlink at %s does not exist', docker_config_path)
+
+        conf = get_worker_config()
+        if os.path.exists(conf.iib_docker_config_template):
+            log.debug(
+                'Copying the Docker config from %s to %s',
+                conf.iib_docker_config_template,
+                docker_config_path,
+            )
+            shutil.copy2(conf.iib_docker_config_template, docker_config_path)
```

Comment:
Why copy if the file is always re-written a few lines below. Could you just read from `conf.iib_docker_config_template` and write to `docker_config_path` once after modifications?

Reply:
Good point. I'll fix that.